src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1009,12 +1009,12 @@ FixedwingPositionControl::control_position(const hrt_abstime &now, const Vector2
 			float target_airspeed = calculate_target_airspeed(mission_airspeed, ground_speed);
 
 			if (_param_fw_use_npfg.get()) {
-				_npfg.setAirspeedNom(target_airspeed);
-				_npfg.setAirspeedMax(_param_fw_airspd_max.get());
+				_npfg.setAirspeedNom(target_airspeed * _eas2tas);
+				_npfg.setAirspeedMax(_param_fw_airspd_max.get() * _eas2tas);
 				_npfg.navigateLoiter(curr_wp, curr_pos, loiter_radius, loiter_direction, ground_speed, _wind_vel);
 				_att_sp.roll_body = _npfg.getRollSetpoint();
 				_att_sp.yaw_body = _npfg.getBearing();
-				target_airspeed = _npfg.getAirspeedRef();
+				target_airspeed = _npfg.getAirspeedRef() / _eas2tas;
```

Comment:
Maybe I am missing something, but shouldn't this be true airspeed to equivalent airspeed and not eas2tas?

It seems that the same conversion is being applied twice to `target_airspeed`

Reply:
All airspeed values in fw pos ctrl currently are equivalent airspeed (target_airspeed, mission_airspeed, etc). The TECS wrapper does its own eas2tas conversion internally. NPFG considers true airspeed, so the equivalent airspeed target needs to be converted to TAS on input to airsp. nom and max. The output airspeed reference from NPFG then is also TAS, so it needs to be converted back (hence the division) before getting input into the TECS wrapper or further augmented in the existing code.

Reply:
@tstastny Your right, I thought all of them were getting multiplied by `_eas2tas` while now I see that one is multiplied and one is divided!